src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/zip/ZipStepExecution.java

```diff
@@ -120,15 +122,20 @@ protected Void run() throws Exception {
     static class ZipItFileCallable extends MasterToSlaveFileCallable<Integer> {
         final FilePath zipFile;
         final String glob;
+        final boolean overwrite;
 
-        public ZipItFileCallable(FilePath zipFile, String glob) {
+        public ZipItFileCallable(FilePath zipFile, String glob, boolean overwrite) {
             this.zipFile = zipFile;
             this.glob = StringUtils.isBlank(glob) ? "**/*" : glob;
+            this.overwrite = overwrite;
         }
 
         @Override
         public Integer invoke(File dir, VirtualChannel channel) throws IOException, InterruptedException {
-            String canonicalZip = new File(zipFile.getRemote()).getCanonicalPath();
+            String canonicalZip = zipFile.getRemote();
+            if (overwrite && !Files.deleteIfExists(Paths.get(canonicalZip))) {
```

Comment:
So now it throws the exception (and fails) if the zip does not exist and overwrite=true. Was that intended?

My case: I need the file to be overwritten when it exists + be created when it does not. That used to work without specifying the overwrite flag (which wasn't there obviously)

Reply:
No, that sounds a bit weird indeed. :thinking:

Reply:
This seems like a bug

Reply:
I am thinking the only workaround for now is to check if file exists before attempting to zip.